File: leetcode_js/tests/twoSum.test.js
const { twoSum } = require('../src/twoSum');

test('1', () => {
  const nums = [2, 7, 11, 15];
  const target = 9;
  const output = [0, 1];

  expect(twoSum(nums, target)).toEqual(output);
});

test('2', () => {
  const nums = [3, 2, 4];
  const target = 6;
  const output = [1, 2];

  expect(twoSum(nums, target)).toEqual(output);
});

test('3', () => {
  const nums = [3, 3];
  const target = 6;
  const output = [0, 1];

  expect(twoSum(nums, target)).toEqual(output);
});

File: leetcode_go/problems/0001_two_sum/two_sum_test.go
package problem0001

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

type input struct {
	nums   []int
	target int
}

type question struct {
	input  input
	output []int
}

func Test_two_sum(t *testing.T) {
	questions := map[string]question{
		"1": {
			input: input{
				nums:   []int{2, 7, 11, 15},
				target: 9,
			},
			output: []int{0, 1},
		},
		"2": {
			input: input{
				nums:   []int{3, 2, 4},
				target: 6,
			},
			output: []int{1, 2},
		},
		"3": {
			input: input{
				nums:   []int{3, 3},
				target: 6,
			},
			output: []int{0, 1},
		},
	}

	for name, question := range questions {
		t.Run(name, func(t *testing.T) {
			result := twoSum(question.input.nums, question.input.target)
			assert.Equal(t, question.output, result)
		})
	}
}
